Code challenge first phase

Antonxx opened the pull request.

callebjorkell commented on .idea/encodings.xml:
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
gitignore all files in `.idea`. No project files should be commited.

callebjorkell commented on src/main/java/com/codechallenge/rest/Service/MessageService.java:
```diff
@@ -0,0 +1,9 @@
+
+import com.codechallenge.model.Message;
+
+public interface MessageService {
```
This is not a service. This is a DTO. Rename accordingly.

callebjorkell commented on src/main/java/com/codechallenge/rest/Resource/MessageResource.java:
```diff
@@ -0,0 +1,32 @@
+    @Path("/names/{name}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.TEXT_PLAIN)
+    public MessageService nameService(@PathParam("name") String name){
```
Naming: Usually, you would have the methods called something like "createMessage" or "getMessage" or something similar. This method name doesn't really describe the "action".

callebjorkell commented on src/main/webapp/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,10 @@
+Manifest-Version: 1.0
```
This file is generated, right? Please remove any generated code from the repository.

callebjorkell commented:
So overall, you need to rename the current `Service` class, and then create a real `MessageService` that is a factory for that DTO. This could have a method eg. `createMessage` to produce the message in some way. The resource should just call this class and then return whatever value the service produces.

You should also fix up the commits according to http://chris.beams.io/posts/git-commit/ and squash them (for example http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html ) to clean up the commit history.

callebjorkell commented:
... also remove any intellij project files and/or other generated files.

callebjorkell commented on src/main/java/com/codechallenge/rest/Service/MessageService.java:
```diff
@@ -0,0 +1,21 @@
+package com.codechallenge.rest.Service;
```
Naming: packages are all lower case
http://www.oracle.com/technetwork/java/codeconventions-135099.html

callebjorkell commented on src/main/java/com/codechallenge/rest/Resource/MessageResource.java:
```diff
@@ -0,0 +1,39 @@
+
+
+    @Autowired
+    private MessageService messageService;
```
Overall, you want to use constructor injection here instead. This because of testability that will become apparent later on :smile: 

i.e. instead of
```java
@Autowired private MessageService messageService;
```
you would do:
```java
private final MessageService messageService;

@Autowired
public MessageResource(MessageService messageService) {
   this.messageService = messageService;
}

callebjorkell commented on src/main/java/com/codechallenge/rest/Service/MessageService.java:
```diff
@@ -0,0 +1,21 @@
+    public MessageDTO createMessage(String name) {
+        name = "Hello " + name;
+        message.setContent(name);
+        return message;
```
This is leaking objects. Since you return the same object to everyone calling the method, as soon as you change it, it will change for all the callers and override the message content.

In other words, don't instantiate this in the constructor, but create a new `MessageDTO` for every invocation of the `createMessage` method. This will also make the service stateless which is a nice bonus :wink:

Antonxx commented:
Code challenge phase one

 -Fixed the problem of directory names squashing all the commits. 
 -Implemented your comments about MessageService and MessageResource classes.

@callebjorkell

callebjorkell commented:
Just remember to put that comment into the description for the next one PR :yum:

callebjorkell commented:
:baby_chick:

callebjorkell commented:
Actually, please re-open this against your own branch (antonio) instead of master. I'll be resetting master back to the real state.

callebjorkell commented:
Also please squash ALL the commits.
